cns/restserver/restserver.go

```diff
@@ -1400,61 +1351,80 @@ func (service *HTTPRestService) detachNetworkContainerFromNetwork(w http.Respons
+			Message:    "[Azure CNS] Error. " + operation + "ContainerToNetwork did not receive a POST."}
+	}
 	if req.Containerid == "" {
-		returnCode = DockerContainerNotSpecified
-		returnMessage = fmt.Sprintf("[Azure CNS] Error. Containerid is empty")
-	} else if req.NetworkContainerid == "" {
-		returnCode = NetworkContainerNotSpecified
-		returnMessage = fmt.Sprintf("[Azure CNS] Error. NetworkContainerid is empty")
-	} else {
-		switch r.Method {
-		case "POST":
-			// try to get the saved nc state if it exists
-			service.lock.Lock()
-			existing, ok := service.state.ContainerStatus[cns.SwiftPrefix+req.NetworkContainerid]
-			service.lock.Unlock()
-			if !ok {
-				returnMessage = fmt.Sprintf("[Azure CNS] Error. Network Container %s does not exist.", req.NetworkContainerid)
-				returnCode = NotFound
-			} else {
-				var podInfo cns.KubernetesPodInfo
-				err := json.Unmarshal(existing.CreateNetworkContainerRequest.OrchestratorContext, &podInfo)
-				if err != nil {
-					returnCode = UnexpectedError
-					returnMessage = fmt.Sprintf("Unmarshalling orchestrator context failed with error %+v", err)
-				} else {
-					nc := service.networkContainer
-					pluginBinPath, _ := service.GetOption(acn.OptCNIPath).(string)
-					configPath, _ := service.GetOption(acn.OptCNIConfigFile).(string)
-					netPluginConfig := networkcontainers.NewNetPluginConfiguration(pluginBinPath, configPath)
-					if err = nc.Detach(podInfo.PodName, podInfo.PodNamespace, req.Containerid, netPluginConfig); err != nil {
-						returnMessage = fmt.Sprintf("[Azure CNS] Error. detachNetworkContainerFromNetwork failed %+v", err.Error())
-						returnCode = UnexpectedError
-					}
-				}
-			}
+		return cns.Response{
+			ReturnCode: DockerContainerNotSpecified,
+			Message:    "[Azure CNS] Error. Containerid is empty"}
+	}
+	if req.NetworkContainerid == "" {
+		return cns.Response{
+			ReturnCode: NetworkContainerNotSpecified,
+			Message:    "[Azure CNS] Error. NetworkContainerid is empty"}
+	}
 
-		default:
-			returnMessage = "[Azure CNS] Error. detachNetworkContainerFromNetwork did not receive a POST."
-			returnCode = InvalidParameter
+	service.lock.Lock()
+	existing, ok := service.state.ContainerStatus[cns.SwiftPrefix+req.NetworkContainerid]
+	service.lock.Unlock()
+	if !ok {
+		return cns.Response{
+			ReturnCode: NotFound,
+			Message:    fmt.Sprintf("[Azure CNS] Error. Network Container %s does not exist.", req.NetworkContainerid)}
+	}
+
+	returnCode := 0
+	returnMessage := ""
+	switch service.state.OrchestratorType {
+	case cns.Batch:
+		var podInfo cns.KubernetesPodInfo
+		err := json.Unmarshal(existing.CreateNetworkContainerRequest.OrchestratorContext, &podInfo)
+		if err != nil {
+			returnCode = UnexpectedError
+			returnMessage = fmt.Sprintf("Unmarshalling orchestrator context failed with error %+v", err)
+		} else {
+			nc := service.networkContainer
+			netPluginConfig := service.getCNIPluginDetails()
+			switch operation {
+			case "Attach":
```

Comment:
same here..move attach, detach to constants

Reply:
done